extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/LiquibaseProcessor.java

```diff
@@ -0,0 +1,298 @@
+import io.quarkus.arc.deployment.GeneratedBeanBuildItem;
+import io.quarkus.deployment.Capabilities;
+import io.quarkus.deployment.annotations.BuildProducer;
+import io.quarkus.deployment.annotations.BuildStep;
+import io.quarkus.deployment.annotations.ExecutionTime;
+import io.quarkus.deployment.annotations.Record;
+import io.quarkus.deployment.builditem.CapabilityBuildItem;
+import io.quarkus.deployment.builditem.FeatureBuildItem;
+import io.quarkus.deployment.builditem.GeneratedResourceBuildItem;
+import io.quarkus.deployment.builditem.ServiceStartBuildItem;
+import io.quarkus.deployment.builditem.nativeimage.*;
+import io.quarkus.liquibase.runtime.LiquibaseBuildTimeConfig;
+import io.quarkus.liquibase.runtime.LiquibaseProducer;
+import io.quarkus.liquibase.runtime.LiquibaseRecorder;
+import io.quarkus.liquibase.runtime.LiquibaseRuntimeConfig;
+import io.quarkus.liquibase.runtime.graal.LiquibaseServiceLoader;
+import liquibase.changelog.ChangeLogParameters;
+import liquibase.changelog.ChangeSet;
+import liquibase.changelog.DatabaseChangeLog;
+import liquibase.exception.LiquibaseException;
+import liquibase.parser.ChangeLogParser;
+import liquibase.parser.ChangeLogParserFactory;
+import liquibase.resource.ClassLoaderResourceAccessor;
+import liquibase.servicelocator.ServiceLocator;
+
+class LiquibaseProcessor {
+
+    private static final Logger LOGGER = Logger.getLogger(LiquibaseServiceLoader.class);
+
+    /**
+     * The change log parser factory singleton used for the liquibase service classpath scanner
+     */
+    private static final ChangeLogParserFactory CHANGE_LOG_PARSER_FACTORY = ChangeLogParserFactory.getInstance();
+
+    /**
+     * Liquibase build time configuration
+     */
+    LiquibaseBuildTimeConfig liquibaseBuildConfig;
+
+    @BuildStep
+    CapabilityBuildItem capability() {
+        return new CapabilityBuildItem(Capabilities.LIQUIBASE);
+    }
+
+    @BuildStep
+    void reflection(BuildProducer<ReflectiveClassBuildItem> reflective,
+            BuildProducer<NativeImageResourceBuildItem> resourceProducer,
+            BuildProducer<GeneratedResourceBuildItem> generatedResourceProducer,
+            BuildProducer<RuntimeInitializedClassBuildItem> runtimeInitialized) {
+
+        runtimeInitialized.produce(new RuntimeInitializedClassBuildItem("liquibase.util.StringUtils"));
+        runtimeInitialized.produce(new RuntimeInitializedClassBuildItem("liquibase.servicelocator.ServiceLocator"));
+        runtimeInitialized.produce(new RuntimeInitializedClassBuildItem("liquibase.diff.compare.CompareControl"));
+
+        //Classes that use reflection
+        reflective.produce(new ReflectiveClassBuildItem(true, false, false,
+                "com.sun.org.apache.xerces.internal.impl.dv.xs.SchemaDVFactoryImpl",
+                "com.sun.org.apache.xerces.internal.jaxp.SAXParserFactoryImpl",
+                "com.sun.org.apache.xerces.internal.jaxp.DocumentBuilderFactoryImpl"));
```

Comment:
It seems to me that this should belong to a jaxp extension

Reply:
@gastaldi should I put jaxb-runtime extension as dependency?
Currently we have only this for the liquibase-runtime
```
        <dependency>
            <groupId>org.jboss.spec.javax.xml.bind</groupId>
            <artifactId>jboss-jaxb-api_2.3_spec</artifactId>
        </dependency>
```

Reply:
Not sure if that belongs to jaxb, perhaps a separate (unlisted) jaxp extension containing this code and the jaxp deps but I'm unsure if it's worth the effort

Reply:
If you need the JAXB API, then it probably makes sense to add the JAXB extension.

Reply:
These classes don't seem to belong to JAXB, but yes, if the JAXB API is used the extension should be added as @gsmet mentioned

Reply:
Ok, I add to the liquibase-runtime the jaxb extension